### PR TITLE
Detect if running in browser context

### PR DESCRIPTION
**Pull request by ebisbe**
Solves #96 .
Tested in local

### CODE REVIEW

**Comment by ebisbe** [comment]
While testing I had an issue with: 
```
'createGtm' is not exported by ../vue-gtm/dist/index.js, imported by src/main.js
file: /Volumes/Data/Development/skedr/landing/src/main.js:6:9
4: import { routes } from "./routes"
5: import VueGtag from "vue-gtag"
6: import { createGtm } from "@gtm-support/vue-gtm"
            ^
7: //import VueHotjar from "vue-hotjar-next"
Error: 'createGtm' is not exported by ../vue-gtm/dist/index.js, imported by src/main.js
    at error (/Volumes/Data/Development/skedr/landing/node_modules/rollup/dist/shared/rollup.js:158:30)
    at Module.error (/Volumes/Data/Development/skedr/landing/node_modules/rollup/dist/shared/rollup.js:12382:16)
    at Module.traceVariable (/Volumes/Data/Development/skedr/landing/node_modules/rollup/dist/shared/rollup.js:12767:29)
    at ModuleScope.findVariable (/Volumes/Data/Development/skedr/landing/node_modules/rollup/dist/shared/rollup.js:11559:39)
    at ReturnValueScope.findVariable (/Volumes/Data/Development/skedr/landing/node_modules/rollup/dist/shared/rollup.js:6930:38)
    at ChildScope.findVariable (/Volumes/Data/Development/skedr/landing/node_modules/rollup/dist/shared/rollup.js:6930:38)
    at Identifier.bind (/Volumes/Data/Development/skedr/landing/node_modules/rollup/dist/shared/rollup.js:6419:40)
    at CallExpression.bind (/Volumes/Data/Development/skedr/landing/node_modules/rollup/dist/shared/rollup.js:5025:23)
    at CallExpression.bind (/Volumes/Data/Development/skedr/landing/node_modules/rollup/dist/shared/rollup.js:9396:15)
    at CallExpression.bind (/Volumes/Data/Development/skedr/landing/node_modules/rollup/dist/shared/rollup.js:5021:31) {
  code: 'MISSING_EXPORT',
  url: 'https://rollupjs.org/guide/en/#error-name-is-not-exported-by-module',
  ```
and I solved using:
```
export { createGtm }
```

But I'm not sure it was just in the testing phase with the bundle or will error also in the build phase...

**Comment by Shinigami92** [comment]
Seems you have not activated `allow changes by maintainer` :slightly_frowning_face:  
Could you do me a favor any copy over my draft https://github.com/gtm-support/vue-gtm/pull/98/files into this PR?
I added a test and I moved the condition one step up.

I would like that you get the credits / commit mention of this PR :slightly_smiling_face: 
Then I can delete my draft

Also please have a look over the `ssg-mode.test.ts` file and check if we should change anything there.

---

One additional question:
I found out that `app.mount('#app')` also fails in "SSG"-Mode :thinking:, so how do you bypass this?

**Comment by Shinigami92** [comment]
Oh, in addition you should make a `rebase` and then use `pnpm`. I switch _slowly_ all my projects to `pnpm` but this project wasn't touched for a while.

**Comment by ebisbe** [comment]
Sorry I didn't know that option existed.

>One additional question:
I found out that app.mount('#app') also fails in "SSG"-Mode 🤔, so how do you bypass this?

 For prerendering I'm using https://github.com/antfu/vite-ssg so I'm not using the options in the Vite directly. That project has a few added benefits that Vite didn't have due to being framework agnostic. 
 
Haven't used `pnpm` but I'll try 👍🏼

**Comment by Shinigami92** [comment]
Ah okay, I just didn't know ViteSSG / how it works :slightly_smiling_face:

**Comment by ebisbe** [comment]
Have a look in https://github.com/Skedr/landing It's the landing page of my project. There's no fancy stuff so easy to see.

**Comment by ebisbe** [comment]
I haven't used TS and I have an error it not happened before when I tried the tests... 

```
ERROR: src/index.ts:150:16 - error TS2664: Invalid module name in augmentation, module '@vue/runtime-core' cannot be found.

150 declare module '@vue/runtime-core' {
                   ~~~~~~~~~~~~~~~~~~~
```
All tests fail now :O

I've made an install with `pnpm` I guess I delete the `yarn.lock` and add the `pnpm-lock.yaml` right?

**Comment by Shinigami92** [comment]
Cause the main branch is now newer than yours

https://github.com/gtm-support/vue-gtm/blob/eb55407f4faf6bf2f4685511e57499d339bc09f0/src/index.ts#L147

**Comment by Shinigami92** [comment]
Seems you copied it with one indent offset
Please run `pnpm run format` and I think we are good to go

**Comment by ebisbe** [comment]
I haven't reviewed the tests :/

**Comment by Shinigami92** [comment]
> I haven't reviewed the tests :/

You are right :+1: I will give you time to do that

File: tests/ssg-mode.test.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+/**
+ * @jest-environment node
```
Comment (Shinigami92) [inline tests/ssg-mode.test.ts:2]: @ebisbe The magic line is this one
This tells jest that in this file stuff like `window` and `document` doesn't exist :slightly_smiling_face: 
So now we should check whatever we want to check

Currently I just copied the normal tests over and changed them a bit, but I'm not sure if this is sufficient

**Comment by ebisbe** [comment]
I've left the only relevant test. If you remove the new `if()` clause that test will return 'ReferenceError: document is not defined`

**Review by Shinigami92 — APPROVED** [review]

**Comment by Shinigami92** [comment]
I will merge it tomorrow or late this evening
You have also some time this way to change anything if needed

**Comment by Shinigami92** [comment]
Released `@gtm-support/vue-gtm@1.2.3`, please test it 🙂

**Comment by ebisbe** [comment]
> Released `@gtm-support/vue-gtm@1.2.3`, please test it 🙂

Everything is working. Thanks!

**Pull request merged**